packages/components/src/button/test/index.js

```diff
@@ -12,172 +13,197 @@ import { plusCircle } from '@wordpress/icons';
 /**
  * Internal dependencies
  */
-import ButtonWithForwardedRef, { Button } from '../';
-import { VisuallyHidden } from '../../visually-hidden';
+import Button from '../';
+
+jest.mock( '../../icon', () => () => <div data-testid="test-icon" /> );
+jest.mock( '../../tooltip', () => ( { text, children } ) => (
+	<div data-testid="test-tooltip" title={ text }>
+		{ children }
+	</div>
+) );
+jest.mock( '../../visually-hidden', () => ( {
+	__esModule: true,
+	VisuallyHidden: ( { children } ) => (
+		<div data-testid="test-visually-hidden">{ children }</div>
+	),
+} ) );
```

Comment:
Yeah, it's a nice way to separate concerns and make sure that we're not stepping into the testing grounds of another component.

Reply:
FWIW I've left the `icon` mock because I think it helps reduce noise in the tests and actually improves the test quality, vs trying to figure out how to query an SVG that's `aria-hidden`.

Reply:
Not sure if this is the most correct conversation to reply to, but for reference this is a convo that we had on the difficulties when testing `VisuallyHidden`:, where we ended up using snapshot testing for it: https://github.com/WordPress/gutenberg/pull/42403#discussion_r922161765


And this is a conversation where we used the `selector` option to make sure that the queried element had a certain `aria` attribute : https://github.com/WordPress/gutenberg/pull/42403#discussion_r932163804